brave-core/src/main/java/com/github/kristofa/brave/InheritableState.java

```diff
@@ -0,0 +1,16 @@
+package com.github.kristofa.brave;
+
+import com.github.kristofa.brave.internal.Nullable;
+import com.twitter.zipkin.gen.Span;
+
+/**
+ * Interface defining the behavior for inheriting state from {@link ServerClientAndLocalSpanState} implementations.
+ */
+public interface InheritableState {
```

Comment:
hmm... let's not make this public. the current approach is a hack, and if we are adding a new public interface it should be something that allows us to use a real context like the recently extracted and dependency free https://github.com/grpc/grpc-java/blob/master/context/src/main/java/io/grpc/Context.java